libs/utils/src/lock_file.rs

```diff
@@ -0,0 +1,82 @@
+//! A module to allow processes to lock on specific FS file exclusive.
+//! Could be used to forbid multiple processes to start in the same directory.
+//!
+//! Note that locked files are done via [`fcntl::flock`] which means that
+//! * files are locked only for taking another lock on them: all other operations are still possible on files.
+//! For example, other process could do `rm file` and that would be allowed.
+//! * if no file exists (after `rm` in another process), the process that "locked" the file still runs and does not error
+//! new procesess are able to start in the same directory, as they will be creating another file to lock.
```

Comment:
Minor editorialization:
```suggestion
//! A module to create and read lock files. A lock file ensures that only one
//! process is running at a time, in a particular directory.
//!
//! File locking is done using [`fcntl::flock`], which means that holding the
//! lock on file only prevents acquiring another lock on it; all other
//! operations are still possible on files. Other process can still open, read,
//! write, or remove the file, for example.  If the file is removed while a
//! process is holding a lock on it, the process that holds the lock does not
//! get any error or notification. Furthermore, you can create a new file with
//! the same name and lock the new file, while the old process is still
//! running. Deleting the lock file while the locking process is still running
//! is a bad idea!
```